src/sign/index.ts

```diff
@@ -47,8 +47,9 @@ export default class Client {
 
   async sign(web3: Web3Provider | Wallet, address: string, message, types) {
     let signer;
-    if (web3 instanceof Wallet) signer = web3;
-    if (web3 instanceof Web3Provider) signer = web3.getSigner();
+    // @ts-ignore
+    if (web3?.getSigner) signer = web3.getSigner();
+    else signer = web3;
```

Comment:
Could be simplified to
```suggestion
  const signer = web3?.getSigner ? web3.getSigner() : web3;
```